services-turf/src/commonMain/kotlin/org/maplibre/geojson/util/Measurement.kt

```diff
@@ -0,0 +1,551 @@
+package org.maplibre.geojson.util
+
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import org.maplibre.geojson.model.Feature
+import org.maplibre.geojson.model.LineString
+import org.maplibre.geojson.model.MultiLineString
+import org.maplibre.geojson.model.MultiPolygon
+import org.maplibre.geojson.model.Point
+import org.maplibre.geojson.model.Polygon
+import kotlin.math.PI
+import kotlin.math.asin
+import kotlin.math.atan2
+import kotlin.math.cos
+import kotlin.math.max
+import kotlin.math.pow
+import kotlin.math.sin
+import kotlin.math.sqrt
+
+/**
+ * Calculates the length of a [LineString] in the specified units.
+ *
+ * @param unit The unit of measurement to use.
+ * @return The length of the line in the specified units.
+ */
+fun LineString.length(unit: MeasureUnit): Double = points.length(unit)
+
+/**
+ * Calculates the total length of a [MultiLineString] in the specified units.
+ *
+ * @param unit The unit of measurement to use.
+ * @return The combined length of all line strings in the specified units.
+ */
+fun MultiLineString.length(unit: MeasureUnit): Double =
+    lineStrings.sumOf { lineString -> lineString.length(unit) }
+
+/**
+ * Calculates the perimeter of a [Polygon] in the specified units, including holes.
+ *
+ * @param unit The unit of measurement to use.
+ * @return The total perimeter of the polygon in the specified units.
+ */
+fun Polygon.length(unit: MeasureUnit): Double =
+    outerLineStringRing.length(unit) +
+            holeLineStringRings.sumOf { line -> line.length(unit) }
+
+/**
+ * Calculates the total perimeter of a [MultiPolygon] in the specified units, including holes.
+ *
+ * @param unit The unit of measurement to use.
+ * @return The total perimeter of all polygons in the specified units.
+ */
+fun MultiPolygon.length(unit: MeasureUnit): Double =
+    polygons.sumOf { polygon -> polygon.length(unit) }
+
+/**
+ * Calculates the length of a list of [Point]s as a line in the specified units.
+ *
+ * @param unit The unit of measurement to use.
+ * @return The length of the line formed by the points in the specified units.
+ */
+private fun List<Point>.length(unit: MeasureUnit): Double {
+    return drop(1)
+        .mapIndexed { index, point ->
+            // Using unmodified index for previous point is working,
+            // because we drop the first point
+            get(index).distanceTo(point, unit)
+        }
+        .sum()
+}
+
+/**
+ * Finds the midpoint between this [Point] and another [Point].
+ *
+ * @param to The other point.
+ * @return The midpoint as a [Point].
+ */
+fun Point.findMidpoint(to: Point): Point {
+    val dist = this.distanceTo(to, MeasureUnit.MILES)
+    val heading = this.bearingTo(to)
+    return this.destination(dist / 2, heading, MeasureUnit.MILES)
+}
```

Comment:
Hmm, other functions are not named something with `find...` (findLength, findBearingTo, ...), but of course, `poin1.midpoint(point2)` reads odd, true. I think a non-extension-function would be more expressive: `midpoint(point1, point2)`

```suggestion
fun midpoint(p1: Point, p2: Point): Point {
    val dist = p1.distanceTo(p2, MeasureUnit.RADIANS)
    val heading = p1.bearingTo(p2)
    return p1.destination(dist / 2, heading, MeasureUnit.RADIANS)
}
```